### PR TITLE
[Metrics] Stub out base components

### PR DESCRIPTION
**Pull request by ncooke3**
This is the initial commit for the new `metrics-main` branch. It stubs out the base components for metric tracking implementation.

I have a series of small PR's queued up.

### CODE REVIEW

**Comment by ncooke3** [comment]
The `check` workflow seemed to flake... 🤔

**Comment by paulb777** [comment]
Might need a PR like https://github.com/google/GoogleUtilities/pull/70

File: GoogleDataTransport/GDTCORLibrary/GDTCORMetricsController.m [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Google LLC
```
Comment (chliangGoogle) [inline GoogleDataTransport/GDTCORLibrary/GDTCORMetricsController.m:2]: 2022

File: GoogleDataTransport/GDTCORLibrary/Internal/GDTCOREventDropReason.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+// TODO(nickcooke): Document.
+typedef NS_ENUM(NSInteger, GDTCOREventDropReason) {
```
Comment (chliangGoogle) [inline GoogleDataTransport/GDTCORLibrary/Internal/GDTCOREventDropReason.h:20]: is this part of the proto defined in the design doc? In that case, are we supposed to use nanopb to present them?

**Review by chliangGoogle — APPROVED** [review]

File: GoogleDataTransport/GDTCORLibrary/Internal/GDTCOREventDropReason.h (hunk 2 above)
Comment (ncooke3) [reply GoogleDataTransport/GDTCORLibrary/Internal/GDTCOREventDropReason.h:20]: Yes, we will use nanopb to represent them but I defined this enum so I have a native objc type for passing to API's like: 

https://github.com/google/GoogleDataTransport/blob/8e1828d0b9d6d85b1be4bd6edf060ecb14c546fa/GoogleDataTransport/GDTCORLibrary/GDTCORMetricsController.m#L39-L41

I will add the proto in future PR.

**Pull request merged**